src/Test/DefaultTest.php

```diff
@@ -53,20 +52,45 @@ public function execute(array $sourceData, array $destinationData): bool
     }
 
     /**
-     * @param  array<string, numeric-string|array<string, numeric-string>>|numeric-string  $data
-     * @param  array<string>  $fields
+     * Recursively checks if an offset exists in a nested array.
+     *
+     * This function is designed to handle nested arrays and verify the existence of
+     * a specified offset (string key) within them. It performs recursive checks on
+     * nested arrays to ensure that the offset exists at each level.
+     *
+     * @param array<string, numeric-string|array<string, numeric-string>>|numeric-string $data
+     *     The data array to check for the offset in.
+     * @param array<string> $fields
+     *     An array of string keys representing the path to the desired offset.
      *
      * @return bool
+     *     Returns `true` if the specified offset exists in the nested array,
+     *     `false` otherwise.
+     *
+     * Used a combination of type hinting and explicit checks to assist PHPStan
+     * in understanding the types involved and resolving static analysis errors.
      */
     private function issetField(mixed $data, array $fields): bool
     {
+        // Check if $data is an array
+        if (!is_array($data)) {
```

Comment:
Why not change the type from mixed to array? This will make sure this code is never called if $data is not an array.

Reply:
@hussainweb Thank you for your feedback on changing the type from mixed to array. I agree and appreciate that it would be a good idea to change the type from mixed to array. This would make sure that the code is never called if $data is not an array and not sure what causes for its not originally addressed by the maintainer.

However, there is one thing i wanted to incorporate that, if we change the type to array, then we will need to handle the case where $data is an empty array. Currently, our code will return true if $data is an empty array. This is because an empty array is still an array, and the key that we are checking for may exist in the empty array.
If we want to change the behavior of this function, then we need to also add the following check to check if $data is not empty.
```
 // Check if $data is an array and is not empty
 if (!is_array($data) || empty($data)) {
  return false;
 }
```
 I'll update and push!

Reply:
That felt complicated. :)

> if we change the type to array, then we will need to handle the case where $data is an empty array

That's true. Do that if you want to. This is not relevant to the discussion on changing the type. `is_array` will still return true if it is an empty array. So the behaviour won't change if we change the parameter type to `array` and remove these lines.

> and the key that we are checking for may exist in the empty array

How is that possible?

> If we want to change the behavior of this function, then we need to also add the following check to check if $data is not empty.

Two things here. First, I am not asking to change the behaviour of this function at all. If we change the type, the behaviour will change but that is for the caller, not the function. I mean, if someone tries to pass anything but an array (even an empty array), it will throw an error.

Second, we don't need `is_array` check at all because that will always return true. PHP will make sure it is an array before it calls the function.